### PR TITLE
Add RxJava2 extensions for Observable.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,16 +30,34 @@ val v = when (optional) {
     is Some -> optional.value // Smart Casts Optional to Some and allows you access its value.
     is None -> "fallback"
 }
+
+// Filter only Some values in RxJava2 streams.
```
Comment (ming13) [inline README.md]: Filter only Some values in RxJava 2 Observable

File: koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/rxjava2spec.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.gojuno.koptional.None
+import com.gojuno.koptional.Some
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.context
+import org.jetbrains.spek.api.dsl.it
+
+class rxjava2spec : Spek({
```
Comment (ming13) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/rxjava2spec.kt]: `RxJavaExtensionsSpec`

File: koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/rxjava2spec.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        Observable.just(Some("a"), None, Some("b"), Some("c"), None)
+    }
+
+    context("filterSome") {
+
+        val subscriber by memoized {
+            TestObserver<String>().apply {
+                stream
+                        .filterSome()
+                        .subscribe(this)
```
Comment (ming13) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/rxjava2spec.kt]: I’m surprised this actually works.

Comment (artem-zinnatullin) [inline koptional-rxjava2-extensions/src/test/kotlin/com/gojuno/koptional/rxjava2/rxjava2spec.kt]: U mean Spek? :D

**Pull request merged**